### PR TITLE
Generalize transforms with Primitive

### PR DESCRIPTION
**Pull request by juliohm**
This PR introduces lazy coordinate transforms with primitive geometries.

### CODE REVIEW

File: test/transforms.jl [hunk 1]
```diff
@@ -357,7 +349,6 @@
     f = Translate(T(1), T(1))
     d = cartgrid(10, 10)
     r, c = TB.apply(f, d)
-    @test r isa CartesianGrid
```
Comment (juliohm) [inline test/transforms.jl]: we need to keep these because they are optimizations

File: test/transforms.jl [hunk 2]
```diff
@@ -689,31 +670,25 @@
...
     @test r ≈ Ellipsoid(T.((4, 8, 12)), cart(1, 4, 9))
-    m = TB.revert(f, r, c)
-    @test m isa SimpleMesh
-    @test m ≈ discretize(g)
+    @test discretize(TB.revert(f, r, c)) ≈ discretize(g)
 
     f = Scale(T(2))
     g = Sphere(cart(1, 2), T(3))
     r, c = TB.apply(f, g)
-    @test r isa Sphere
```
Comment (juliohm) [inline test/transforms.jl]: undo

File: test/transforms.jl [hunk 3]
```diff
@@ -689,31 +670,25 @@
...
-    @test r ≈ f(m)
+    @test discretize(r) ≈ f(m)
     @test centroid(r) ≈ f(centroid(g))
-    @test TB.revert(f, r, c) ≈ m
+    @test discretize(TB.revert(f, r, c)) ≈ m
 
     f = Scale(T(1), T(2), T(3))
     g = Sphere(cart(1, 2, 3), T(4))
     r, c = TB.apply(f, g)
-    @test r isa Ellipsoid
```
Comment (juliohm) [inline test/transforms.jl:700]: undo

File: test/transforms.jl [hunk 4]
```diff
@@ -689,31 +670,25 @@
...
     g = Sphere(cart(1, 2), T(3))
     r, c = TB.apply(f, g)
-    @test r isa Sphere
     @test r ≈ Sphere(cart(2, 4), T(6))
     @test TB.revert(f, r, c) ≈ g
 
     f = Scale(T(2))
     g = Sphere(cart(1, 2, 3), T(4))
     r, c = TB.apply(f, g)
-    @test r isa Sphere
```
Comment (juliohm) [inline test/transforms.jl]: undo

File: test/transforms.jl [hunk 5]
```diff
@@ -1322,7 +1277,6 @@
     f = LengthUnit(u"km")
     g = Sphere(cart(0, 0), T(1000))
     r, c = TB.apply(f, g)
-    @test r isa Sphere
```
Comment (juliohm) [inline test/transforms.jl]: undo

File: test/transforms.jl [hunk 6]
```diff
@@ -1332,7 +1286,6 @@
     f = LengthUnit(u"cm")
     g = Ellipsoid(T.((1, 1, 1)), cart(0, 0, 0))
     r, c = TB.apply(f, g)
-    @test r isa Ellipsoid
```
Comment (juliohm) [inline test/transforms.jl]: undo

File: test/transforms.jl [hunk 7]
```diff
@@ -1795,7 +1739,6 @@
     quad = Quadrangle(cart(0, 1, 0), cart(1, 1, 0), cart(1, 0, 0), cart(0, 0, 0))
     repair = Repair(9)
     rquad, cache = TB.apply(repair, quad)
-    @test rquad isa Quadrangle
```
Comment (juliohm) [inline test/transforms.jl]: undo

**Review by eliascarv — APPROVED** [review]

**Pull request merged**